api/broker/broker_controller.go

```diff
@@ -85,94 +97,144 @@ func (ctrl *Controller) addBroker(response http.ResponseWriter, request *http.Re
 }
 
-func (ctrl *Controller) getAllBrokers(response http.ResponseWriter, request *http.Request) error {
+func (c *Controller) getAllBrokers(response http.ResponseWriter, request *http.Request) error {
 	logrus.Debug("Getting all brokers")
-
-	brokers, err := ctrl.BrokerStorage.GetAll()
+	brokers, err := c.BrokerStorage.GetAll()
 	if err != nil {
 		return err
 	}
+	withCatalog := request.FormValue(catalogParam)
+	if strings.ToLower(withCatalog) != "true" {
+		for i := 0; i < len(brokers); i++ {
+			brokers[i].Catalog = nil
+		}
+	}
 
-	type brokerResponse struct {
+	return rest.SendJSON(response, http.StatusOK, struct {
 		Brokers []types.Broker `json:"brokers"`
-	}
-	return rest.SendJSON(response, http.StatusOK, brokerResponse{brokers})
+	}{
+		Brokers: brokers,
+	})
 }
 
-func (ctrl *Controller) deleteBroker(response http.ResponseWriter, request *http.Request) error {
+func (c *Controller) deleteBroker(response http.ResponseWriter, request *http.Request) error {
 	brokerID := getBrokerID(request)
 	logrus.Debugf("Deleting broker with id %s", brokerID)
 
-	err := ctrl.BrokerStorage.Delete(brokerID)
+	err := c.BrokerStorage.Delete(brokerID)
 	err = common.HandleNotFoundError(err, "broker", brokerID)
 	if err != nil {
 		return err
 	}
 	return rest.SendJSON(response, http.StatusOK, map[string]int{})
 }
 
-func (ctrl *Controller) updateBroker(response http.ResponseWriter, request *http.Request) error {
+func (c *Controller) patchBroker(response http.ResponseWriter, request *http.Request) error {
 	brokerID := getBrokerID(request)
-	logrus.Debugf("Updating broker with id %s", brokerID)
+	logrus.Debugf("Updating updateBroker with id %s", brokerID)
 
-	broker := &types.Broker{}
-	if err := rest.ReadJSONBody(request, broker); err != nil {
-		logrus.Error("Invalid request body")
+	updateBroker := &types.Broker{}
+	if err := rest.ReadJSONBody(request, updateBroker); err != nil {
 		return err
 	}
 
-	broker.ID = brokerID
-	broker.UpdatedAt = time.Now().UTC()
+	updateBroker.ID = brokerID
+	updateBroker.UpdatedAt = time.Now().UTC()
 
-	brokerStorage := ctrl.BrokerStorage
-	if broker.Credentials != nil {
-		err := validateBrokerCredentials(broker.Credentials)
+	if updateBroker.Credentials != nil {
+		err := validateBrokerCredentials(updateBroker.Credentials)
 		if err != nil {
 			return types.NewErrorResponse(err, http.StatusBadRequest, "BadRequest")
 		}
 	}
-	err := brokerStorage.Update(broker)
+
+	broker, err := c.BrokerStorage.Get(updateBroker.ID)
+	err = common.HandleNotFoundError(err, "broker", brokerID)
+	if err != nil {
+		return err
+	}
+
+	updateData, err := json.Marshal(updateBroker)
+	if err != nil {
+		return err
+	}
+	if err := json.Unmarshal(updateData, broker); err != nil {
+		return err
+	}
+
+	catalog, err := c.getBrokerCatalog(broker)
+	if err != nil {
+		return types.NewErrorResponse(err, http.StatusBadRequest, "BadRequest")
+	}
+
+	isCatalogModified := !bytes.Equal(broker.Catalog, catalog)
```

Comment:
I thought it would be faster this way (basically if its not modifed we don't provided it to the db query) -I havent benchmarked it though - it  would work without this check as well, too